packages/dds/tree/src/feature-libraries/modular-schema/genericFieldKind.ts

```diff
@@ -66,26 +70,36 @@ export const genericChangeHandler: FieldChangeHandler<GenericChangeset> = {
             const composed: GenericChangeset = [];
             for (const change of changes) {
                 let listIndex = 0;
-                for (const { index, nodeChange } of change) {
+                for (const { index, nodeChange } of change.change) {
+                    const taggedChange = tagChange(nodeChange, change.revision);
                     while (listIndex < composed.length && composed[listIndex].index < index) {
                         listIndex += 1;
                     }
                     const match: GenericChange | undefined = composed[listIndex];
                     if (match === undefined) {
-                        composed.push({ index, nodeChange });
+                        composed.push({ index, nodeChange: composeChildren([taggedChange]) });
                     } else if (match.index > index) {
-                        composed.splice(listIndex, 0, { index, nodeChange });
+                        composed.splice(listIndex, 0, {
+                            index,
+                            nodeChange: composeChildren([taggedChange]),
+                        });
                     } else {
                         composed.splice(listIndex, 1, {
                             index,
-                            nodeChange: composeChildren([match.nodeChange, nodeChange]),
+                            nodeChange: composeChildren([
+                                tagChange(match.nodeChange, match.revision),
```

Comment:
This `match.revision` will always be `undefined` at the moment.
I think we have to options:

A) Set the `GenericChange.revision` field in `if (match === undefined)` and `else if (match.index > index)` branches above as we insert into the `composed` array.

B) Maintain a `Map<NodeChangeset, RevisionTag>` for each change that is added to `composed` and query it instead of using `match.revision`. If we do this, then we can get rid of the the `GenericChange.revision` field entirely, which would be nice.